### PR TITLE
LPD-18657 Fix Image tags being cloned in Product API

### PR DESCRIPTION
**Pull request by joanagrego**
https://liferay.atlassian.net/browse/LPD-18657

### CODE REVIEW

**Comment by joanagrego** [comment]
ci:test:sf

**Review by andrea-ale-sbarra — CHANGES_REQUESTED** [review]
Please make the test that I suggested to make sure the fix work also in this case.
Thank you!

File: modules/apps/commerce/headless/headless-commerce/headless-commerce-admin-catalog-impl/src/main/java/com/liferay/headless/commerce/admin/catalog/internal/resource/v1_0/ProductResourceImpl.java [hunk 1]
```diff
@@ -1048,10 +1048,18 @@ private CPDefinition _updateNestedResources(
...
+
+				if (attachment.getTags() == null) {
+					assetTagNames = transformToArray(
+						_assetTagService.getTags(
+							cpDefinition.getModelClassName(),
+							cpDefinition.getCPDefinitionId()),
+						AssetTag::getName, String.class);
 				}
 
+				serviceContext.setAssetTagNames(assetTagNames);
```
[the remark this answers is not in the document]
Comment (joanagrego) [reply modules/apps/commerce/headless/headless-commerce/headless-commerce-admin-catalog-impl/src/main/java/com/liferay/headless/commerce/admin/catalog/internal/resource/v1_0/ProductResourceImpl.java:1061]: Hi @andrea-ale-sbarra, I see. You're right! I was wondering if we would have problems when trying to set a null value, but it works.

**Comment by joanagrego** [comment]
Hi @andrea-ale-sbarra, I see. You're right! I was wondering if we would have problems when trying to set a null value, but it works. Can you take a look here? 

Thanks for your contribution.

**Review by andrea-ale-sbarra — APPROVED** [review]
LGTM.
Thank you!

**Review by smottal — APPROVED** [review]

**Comment by joanagrego** [comment]
ci:test:relevant

**Comment by smottal** [comment]
ci:report:2547821810

**Comment by smottal** [comment]
ci:test:sf

**Comment by joanagrego** [comment]
ci:test:relevant

**Comment by smottal** [comment]
ci:forward

**Pull request closed without merging**